src/crepe/system/AnimatorSystem.cpp

```diff
@@ -13,12 +12,27 @@ void AnimatorSystem::update() {
 
 	RefVector<Animator> animations = mgr.get_components_by_type<Animator>();
 
-	uint64_t tick = SDLContext::get_instance().get_ticks();
+	double elapsed_time = this->timer.get_current_time();
+
 	for (Animator & a : animations) {
 		if (!a.active) continue;
-		// (10 frames per second)
-		a.curr_row = (tick / 100) % a.row;
-		a.spritesheet.mask.x = (a.curr_row * a.spritesheet.mask.w) + a.curr_col;
-		a.spritesheet.mask = a.spritesheet.mask;
+
+		Animator::Data & ctx = a.data;
+		double frame_duration = 1.0f / ctx.fps;
+
+		int last_frame = ctx.curr_row;
+
+		int cycle_end = (ctx.cycle_end == -1) ? ctx.row : ctx.cycle_end;
+		int total_frames = cycle_end - ctx.cycle_start;
+
+		int curr_frame = static_cast<int>(elapsed_time / frame_duration) % total_frames;
+
+		ctx.curr_row = ctx.cycle_start + curr_frame;
+		ctx.spritesheet.mask.x = ctx.curr_row * ctx.spritesheet.mask.w;
+		ctx.spritesheet.mask.y = (ctx.curr_col * ctx.spritesheet.mask.h);
+
+		if (!ctx.looping && curr_frame == ctx.cycle_start && last_frame == total_frames - 1) {
+			a.active = false;
```

Comment:
Even voor de goede orde: Als een Animator.active false is, betekend het dat de Sprite (één enkele Sprite, niet de gehele SpriteSheet) dan wel wordt getoond maar gewoon niet wordt geanimeerd? Dat is namelijk wat ik zou verwachten.

PS Als ik in de Scene direct de Animator.active op false zet, krijg ik geen Sprite te zien, terwijl ik de aller eerste Sprite (zonder animatie) zou verwachten.

Reply:
dit komt waarschijnlijk omdat in de struct van animator heb je dan niet col of row meegestuurt en die hebben dan geen waardes. ik kan dit oplossen door een constuctor in de struct te zetten waarbij daarin de minimale invulwaardes word aangegeven

Reply:
Ik snap niet echt wat dit te maken heeft met het wel/niet doorsturen van col of row waardes (ik heb het hier over het gedrag wat ontstaat bij het setten/resetten van `active`). Zoals hierboven genoemd, zou ik verwachten dat als een Animator.active false is, het betekend dat de Sprite (één enkele Sprite, niet de gehele SpriteSheet) dan wel wordt getoond maar gewoon niet wordt geanimeerd.

Dus als ik het volgende instel:
![image](https://github.com/user-attachments/assets/fc142a3f-a94b-46f1-896a-d8904d17455f)
Zou ik verwachten dat ik de eerste Sprite (van Column 2) te zien krijg, maar dat is niet het geval.

Reply:
dit komt omdat animator niet eerst heeft gerunt. ik kan dit oplossen